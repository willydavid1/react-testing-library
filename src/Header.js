import React from 'react'

const Header = ({ text }) => {
  return (
    <header>
      <h1 data-testid="h1tag" className="fancy-h1">{text}</h1>
    </header>
  )
}

export default Header
