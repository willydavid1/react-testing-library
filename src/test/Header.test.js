import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Header from '../Header'

afterEach(cleanup) // Unmounts React trees that were mounted with render.

describe('-> Header tests', () => {
  const propText = 'Hello!'
  it('should toMatchSnapshot', () => {
    const { asFragment } = render(<Header text={propText} />)
    expect(asFragment()).toMatchSnapshot()
  })
  
  it('should inserts text in h1', () => {
    const { getByTestId, getByText } = render(<Header text={propText} />)
    expect(getByTestId('h1tag')).toHaveTextContent(propText)
    expect(getByText(propText)).toHaveClass('fancy-h1')
  })
})
