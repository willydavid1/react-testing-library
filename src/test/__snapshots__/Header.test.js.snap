// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-> Header tests should toMatchSnapshot 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="fancy-h1"
      data-testid="h1tag"
    >
      Hello!
    </h1>
  </header>
</DocumentFragment>
`;
